### DISCUSSION TITLE
How do I ingest data to influx from Artillery Tests

### DISCUSSION THREAD

**Discussion by Thilaknath** [artilleryio/artillery#2282] [Artillery]
Hello Team, We host a influx instance to which we want to write our performance execution results from artillery. The documentation doesn't seem to detail much. Can you help us how to set this up?

**Comment by hassy** [comment]
hi @Thilaknath 👋 Artillery's `publish-metrics` plugin supports Influx through Telegraf - have you tried that? https://www.artillery.io/docs/reference/extensions/publish-metrics#influxdbtelegraf

**Reply by Thilaknath** [reply]
@hassy Yes I did read this documentation. Does it imply I would have to be running telgraf on the machine where I invoke the artillery command. As our execution takes place through jenkins which in turn runs it in a dynamic k8s pod. So am not sure how I could go about this whole setup.

**Reply by hassy** [reply]
Telegraf would usually run on the same host as InfluxDB itself. This blog post outlines how it works: https://www.influxdata.com/blog/getting-started-with-sending-statsd-metrics-to-telegraf-influxdb/

Do you know if your InfluxDB is self-hosted or if you're using InfluxDB Cloud?

**Reply by hassy** [reply]
Telegraf/StatsD aside, InfluxDB has an [official Javascript client](https://docs.influxdata.com/influxdb/cloud/api-guide/client-libraries/nodejs/write/), we can look at adding support for that to the `publish-metrics` plugin.

**Reply by Thilaknath** [reply]
That would be very helpful, If we could directly configure details for the influx in our test scripts like dynatrace. We are using a self hosted influxDB

**Reply by InesNi** [reply]
Hi @Thilaknath 👋🏻,

Would the OpenTelemetry reporter maybe work for you for this use case? If I remember correctly you have an OpenTelemetry setup,  maybe you could send the data from Artillery to the Collector and then use an [InfluxDb exporter for the Collector](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/influxdbexporter)?

If that doesn't work, we'd be open to having a short call to discuss your use-case a bit further, in case we need to make `InfluxDb` support better. Feel free to send me a message over at ifazlic@artillery.io !

**Reply by Thilaknath** [reply]
@InesNi The open telemetry collector no longer works at our end since we the provider switched to mTLS. I have dropped progressing on reporting to open telemetry as dynatrace seems to work just fine. But influx integration would be even more ideal.

**Reply by kushal45** [reply]
Hi,
I need to directly integrate influxdb by passing the config {
 server,
 username,
 database,
 passwd
}

Can I do that in the yml file in the plugins section so that load test data is ingested to influx db which is then read by grafana ?
